lib/perl/Genome/Model/Tools/Transcriptome/ErccMapUnaligned.pm

```diff
@@ -0,0 +1,367 @@
+    );
+
+    my $stream = Genome::Sys::ShellPipeline->create(
+        pipe_commands => [$cmd1, $cmd2, $cmd3, $cmd4],
+        redirects => "> $stdout 2> $stderr",
+    );
+
+    $stream->execute
+      or die "[err] Trouble executing remapped bam stream command!\n";
+
+    unless (-e $remapped_bam_path) {
+        die "[err] Couldn't find the remapped bam: $remapped_bam_path \n!";
+    }
+
+    return $remapped_bam_path;
+}
+
+sub index_bam {
+    my ($self, $bam) = @_;
+
+    my $cmd = join(' ', $self->samtools, 'index', "$bam");
+    my $index_file = Path::Class::File->new(
+        $bam->stringify . '.bai'
+    );
+
+    Genome::Sys->shellcmd(
+        cmd => $cmd,
+        input_files => ["$bam"],
+        output_files => ["$index_file"],
+        skip_if_output_is_present => 0
+    );
+}
+
+sub generate_idxstats {
+    my ($self, $bam) = @_;
+
+    my $samtools_idxstats_path = Path::Class::File->new(Genome::Sys->create_temp_file_path());
+
+    my $cmd = join(' ',
+        $self->samtools, 'idxstats',
+        "$bam", 
+        '>', "$samtools_idxstats_path"
+    );
+
+    Genome::Sys->shellcmd(
+        cmd => $cmd,
+        input_files => ["$bam"],
+        output_files => ["$samtools_idxstats_path"],
+        skip_if_output_is_present => 0
+    );
+
+    my $idxstats_hash_ref = Genome::Model::Tools::Sam::Idxstats->parse_file_into_hashref(
+        "$samtools_idxstats_path"
+    );
+
+    return $idxstats_hash_ref;
+}
+
+sub generate_tsvfile {
+    my ($self, $idxstats) = @_;
+
+    my $r_input_file = Genome::Sys->create_temp_file_path();
+
+    my @headers = (
+        'Re-sort ID',
+        'ERCC ID',
+        'subgroup',
+        'ERCC Mix',
+        'concentration (attomoles/ul)',
+        'label',
+        'count',
+    );
+
+    my $writer = Genome::Utility::IO::SeparatedValueWriter->create(
+        output => $r_input_file,
+        headers => \@headers,
+        separator => "\t",
+    );
+
+    my %ercc_control = $self->load_ercc_control_info;
+
+    for my $chr (keys %{$idxstats}) {
+        my $ercc_data = $ercc_control{$chr};
+        unless ($ercc_data) {
+            die('Missing chromosome: '. $chr);
+        }
+        my $concentration = $ercc_data->{'concentration in Mix 1 (attomoles/ul)'};
+        if ($self->ercc_spike_in_mix == 2) {
+            $concentration = $ercc_data->{'concentration in Mix 2 (attomoles/ul)'};
+        }
+        my %data = (
+            'Re-sort ID' => $ercc_data->{'Re-sort ID'},
+            'ERCC ID' => $ercc_data->{'ERCC ID'},
+            'subgroup' => $ercc_data->{'subgroup'},
+            'ERCC Mix' => $self->ercc_spike_in_mix,
+            'concentration (attomoles/ul)' => $concentration,
+            'label' => 'na',
+            'count' => $idxstats->{$chr}{map_read_ct},
+        );
+        $writer->write_one(\%data);
+    }
+    $writer->output->close;
+
+    return Path::Class::File->new($r_input_file);
+}
+
+sub copy_to {
```

Comment:
The `copy_to` functionality is in a newer edition of `Path::Class`.  Our currently installed version doesn't have it.